scheduler/src/cook/mesos/task.clj

```diff
@@ -60,17 +91,26 @@
                        cook-executor? :cook-executor
                        ;; use mesos' command executor by default
                        :else :command-executor)
+        executor (case executor-key
+                   :command-executor :executor/mesos
+                   :container-command-executor :executor/mesos
+                   :cook-executor :executor/cook
+                   :executor/custom)
         data (.getBytes
                (if cook-executor?
                  (json/write-str {"command" (:job/command job-ent)})
                  (pr-str
                    ;;TODO this data is a race-condition
                    {:instance (str (count (:job/instance job-ent)))}))
                "UTF-8")]
+    (when (and (= :executor/cook (:job/executor job-ent))
+               (not= executor-key :cook-executor))
+      (log/warn "Task" task-id "requested to use cook executor, but will be executed using" (name executor-key)))
```

Comment:
When would this happen? If they requested to use cook executor but it is not configured?

Reply:
Yes, when they requested to use cook executor but it is not configured.

Reply:
I wish there was a way to inform users why we switched on them. I don't think it is worth holding up this PR, but something like a log on each job of choices made that are "unexpected" so that a user can get some feedback on what happened.

Reply:
Agree.

Reply:
Have we created an issue for @wyegelwel's suggestion? I'm happy to create one if not